### DISCUSSION TITLE
Data Source refresh stuck in queued state

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#15036] [Help Wanted!]
One of my local data sources got stuck in the **Queued** state when I tried to sync it today, so now I'm unable to click the sync button (it's grayed out).  I've used the sync feature dozens of times before so I'm not sure why it happened this time.  There are no pending or running jobs showing in `/core/data-sources/1/jobs/` or `/core/jobs/`.
I've tried restarting the service with `systemctl restart netbox netbox-rq`. How can I fix this?

<img width="507" alt="image" src="https://github.com/netbox-community/netbox/assets/3991046/bd780da5-265c-4fae-9f04-389d01d62612">

**Comment by llamafilm** [comment]
I fixed it in nbshell:
```
>>> d = DataSource.objects.get(id=1)
>>> d.status
'queued'
>>> d.status = 'completed'
>>> d.full_clean()
Loaded configuration data from cache
Initialized configuration
>>> d.save()
```

**Comment by kvkewn** [comment]
Has anyone else run into this? I just got it whilst testing Data Sources. 
I wonder if it's been looked at by the devs, couldn't find any more mentions.

**Reply by HMohammad2520** [reply]
check the rqworker is running or working properly.

**Comment by akire0ne** [comment]
Not sure if it can help, but on our side, we observed that issue when we had scripts from our datasource that were being executed.
It seems that when they are running, they will prevent the datasource to be synced until script completion.
Check your Job status...

**Reply by HMohammad2520** [reply]
hi 
i think adding more netbox-rq workers will fix your issue,
if you have a heavy environment, multi tasking will help

i do not know how to add new instances :D and i never did
hope you will figure it out

but in my case , the problem was the netbox-rq crashed and i manually started it again
commands:
```bash
systemctl enable netbox-rq.service
service netbox-rq restart
```
hope this helps

**Comment by davama** [comment]
had a similar issue

we were doing mass updates which triggered an event rule which we were testing.
Thankfully the jobs themselves failed because of `ssl verification` error but the background tasks was very full and was not clearing.

Way i cleared it was doing it the dirty way on redis itself.

```
redis-cli KEYS "rq:job:*" | xargs redis-cli DEL
```

copilot had said to do this `Job.objects.all().delete()` in `nbshell` but that didnt work
